modules/service/src/main/scala/lucuma/odb/graphql/mapping/VisitMapping.scala

```diff
@@ -126,4 +130,44 @@ trait VisitMapping[F[_]] extends VisitTable[F]
       selectWithOffsetAndLimit(rOFFSET, rLIMIT, ExecutionEventType, "id", Predicates.executionEvent.id, Predicates.executionEvent.observation.program)
   }
 
+  // WIP.  At the moment the `invoiceHandler` calls the time accounting service
+  // on demand and returns an "invoice" with just the whole execution time.
+  private lazy val invoiceHandler: EffectHandler[F] =
```

Comment:
While this is done on demand at the moment, ultimately the results will be stored (per visit).  In addition to making it a fast/simple lookup for executed visits, if the rules change after the fact then the change won't impact past calculations. Stability is important because otherwise changing the implementation invalidates any manual corrections which may have been applied.